src/shared-components/immersiveModal/crossIconComponent/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react';
+import { useFocusManager } from '@react-aria/focus';
+
+import CrossIcon from '../../../svgs/icons/cross-icon.svg';
+import { CrossIconContainer } from '../style';
+
+type CrossIconComponentProps = {
+  showDesktopHeaderBar: boolean;
+  onClick: () => void;
+};
+
+/**
+ * We wrap the second of the CrossIcons we render as a React component
+ * in order to use the @react-aria/focus useFocusManager hook.
+ *
+ * This allows us to automatically toggle focus state between the two separate
+ * CrossIcons present in the markup. (We keep both rendered for our transitions.)
+ */
+const CrossIconComponent = ({
+  showDesktopHeaderBar,
```

Comment:
[super-nit] - I might name this `isVisible` or some such. I think that's a little bit easier to understand (when the prop becomes not visible, it passes focus back to another focusable thing) by reading this file alone, instead of needing to pass back and forth to the `immersiveModal` file. May not make any difference, as folks are likely to start at the ImmersiveModal anyway, so feel free to disregard.

Reply:
I'm happy to rename things. I just don't typically rename things that are already in place, but when a reviewer suggests it, that's all the approval I need. 😈